slsim/lens_pop.py

```diff
@@ -41,6 +49,10 @@ def __init__(
         :type filters: list of strings or None
         """
         super().__init__(sky_area, cosmo)
+        if source_type == "galaxies" and kwargs_variability is not None:
+            raise ValueError("Extended source cannot have variability.")
+        self.variability_model = variability_model
```

Comment:
do we need this as a self-variable or do we have access to it through other classes?